packages/babel-plugin-proposal-decorators/src/transformer.js

```diff
@@ -1,3 +1,212 @@
-// Not implemented yet
+import { types as t, template } from "@babel/core";
+import splitExportDeclaration from "@babel/helper-split-export-declaration";
+import ReplaceSupers from "@babel/helper-replace-supers";
 
-export default {};
+function prop(key, value) {
+  if (!value) return null;
+  return t.objectProperty(t.identifier(key), value);
+}
+
+function value(body, params = []) {
+  return t.objectMethod("method", t.identifier("value"), params, body);
```

Comment:
I think that ES5 was a good target for ES2015/ES2016 transforms, but now we can output ES6 since users will either only support browsers which support ES6 or compile it down with `preset-env`.